### PR TITLE
[Feature] Add PoolFormer (CVPR'2022)

### PR DESCRIPTION
**Pull request by MengzhangLI**
## Motivation
Support PoolFormer (CVPR'2022)
Paper: [MetaFormer is Actually What You Need for Vision](https://arxiv.org/abs/2111.11418)
Code: https://github.com/sail-sg/poolformer/tree/main/segmentation
Upstream backbone from MMClassification: https://github.com/open-mmlab/mmclassification/tree/master/configs/poolformer 

- [x] Align Inference Results against official repo
- [x] Align Training metric of models to repeat results
- [x] Post-processing: Upload README.md and models & logs.
 
## Results about Aligning Inference metric

Method | Backbone | Pretrain | Iters | mIoU
-- | -- | -- | -- | -- 
Semantic FPN | PoolFormer-S12 | ImageNet-1K | 40K | 37.15
Semantic FPN | PoolFormer-S24 | ImageNet-1K | 40K | 40.27
Semantic FPN | PoolFormer-S36 | ImageNet-1K | 40K | 41.97
Semantic FPN | PoolFormer-M36 | ImageNet-1K | 40K | 42.36
Semantic FPN | PoolFormer-M48 | ImageNet-1K | 40K | 42.72

Thus, the inference metric has been aligned, which means the model structure and test pipeline are both totally correct.

## Results about Aligning Training metric

Method | Backbone | Pretrain | Iters | mIoU (paper) | re-training
-- | -- | -- | -- | --  | --
Semantic FPN | PoolFormer-S12 | ImageNet-1K | 40K | 37.15 | 37.07
Semantic FPN | PoolFormer-S24 | ImageNet-1K | 40K | 40.27 | 40.51
Semantic FPN | PoolFormer-S36 | ImageNet-1K | 40K | 41.97 | 41.81
Semantic FPN | PoolFormer-M36 | ImageNet-1K | 40K | 42.36 | 42.35
Semantic FPN | PoolFormer-M48 | ImageNet-1K | 40K | 42.72 | 42.43

Thus, PoolFormer could be re-implemented in this PR.

## Total Results

| Method | Backbone | Crop Size | pretrain | Batch Size | Lr schd | Mem (GB) | Inf time (fps) | mIoU  | mIoU(ms+flip) | mIoU\*  | mIoU\*(ms+flip) |config | download |
| ------ | -------- | --------- | ----------   | ------ | ------- | -------- | -------------- | ----- | ------------: | ----- | ------------: |-------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| FPN | PoolFormer-S12 | 512x512 | ImageNet-1K | 32     | 40000   | 4.17     | 23.48          | 36.0  | 36.42         | 37.07 | 38.44         | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/poolformer/fpn_poolformer_s12_8x4_512x512_40k_ade20k.py)  | [model](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_s12_8x4_512x512_40k_ade20k/fpn_poolformer_s12_8x4_512x512_40k_ade20k_20220501_115154-b5aa2f49.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_s12_8x4_512x512_40k_ade20k/fpn_poolformer_s12_8x4_512x512_40k_ade20k_20220501_115154.log.json)     |
| FPN | PoolFormer-S24 | 512x512 | ImageNet-1K | 32     | 40000   | 5.47     | 15.74          | 39.35 | 39.73         | 40.36 | 41.08         | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/poolformer/fpn_poolformer_s24_8x4_512x512_40k_ade20k.py)  | [model](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_s24_8x4_512x512_40k_ade20k/fpn_poolformer_s24_8x4_512x512_40k_ade20k_20220503_222049-394a7cf7.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_s24_8x4_512x512_40k_ade20k/fpn_poolformer_s24_8x4_512x512_40k_ade20k_20220503_222049.log.json)     |
| FPN | PoolFormer-S36 | 512x512 | ImageNet-1K | 32     | 40000   | 6.77     | 11.34          | 40.64 | 40.99         | 41.81 | 42.72         | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/poolformer/fpn_poolformer_s36_8x4_512x512_40k_ade20k.py)  | [model](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_s36_8x4_512x512_40k_ade20k/fpn_poolformer_s36_8x4_512x512_40k_ade20k_20220501_151122-b47e607d.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_s36_8x4_512x512_40k_ade20k/fpn_poolformer_s36_8x4_512x512_40k_ade20k_20220501_151122.log.json)     |
| FPN | PoolFormer-M36 | 512x512 | ImageNet-1K | 32     | 40000   | 8.59     | 8.97           | 40.91 | 41.28         | 42.35 | 43.34         | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/poolformer/fpn_poolformer_m36_8x4_512x512_40k_ade20k.py)  | [model](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_m36_8x4_512x512_40k_ade20k/fpn_poolformer_m36_8x4_512x512_40k_ade20k_20220501_164230-3dc83921.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_m36_8x4_512x512_40k_ade20k/fpn_poolformer_m36_8x4_512x512_40k_ade20k_20220501_164230.log.json)     |
| FPN | PoolFormer-M48 | 512x512 | ImageNet-1K | 32     | 40000   | 10.48    | 6.69           | 41.82 | 42.2          | 42.76 | 43.57         | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/poolformer/fpn_poolformer_m48_8x4_512x512_40k_ade20k.py)  | [model](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_m48_8x4_512x512_40k_ade20k/fpn_poolformer_m48_8x4_512x512_40k_ade20k_20220504_003923-64168d3b.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/poolformer/fpn_poolformer_m48_8x4_512x512_40k_ade20k/fpn_poolformer_m48_8x4_512x512_40k_ade20k_20220504_003923.log.json)     |

Note:

- `mIoU` with \* is collected when `ResizeToMultiple` is adopted in `test_pipeline`, so do `mIoU` in logs.

### CODE REVIEW

File: mmseg/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -1333,3 +1333,110 @@ def __repr__(self):
...
+        Args:
+            results (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Resized results, 'img_shape', 'pad_shape', 'scale_factor',
+                'keep_ratio' keys are added into result dict.
+        """
+
+        if 'scale' not in results:
+            self._random_scale(results)
```
Comment (MeowZheng) [inline mmseg/datasets/pipelines/transforms.py:1431]: not implementation `_random_scale`

**Review by MeowZheng — APPROVED** [review]
